reasoner_util/__init__.py

```diff
@@ -185,3 +185,45 @@ def merge_qnodes(qnodes1: dict, qnodes2: dict) -> dict:
         elif qnode_value != new_nodes[qnode_key]:
             raise ValueError("Key exists in both messages but values do not match.")
     return new_nodes
+
+
+def merge_attributes(
+        knode1_attributes: List[dict],
+        knode2_attributes: List[dict],
+        in_place: bool = False
+     ) -> List[dict]:
```

Comment:
```suggestion
    ) -> List[dict]:
```
Do you mean to have the closing parenthesis aligned with the beginning of the function name?

Reply:
@patrickkwang That is what I wanted- however, my flake8 linter yells at me for "continuation line with same indent as next logical line". What should I do here?

Reply:
I think convention here is to align the closing paren with the `def` ie no indent. 

```
def merge_attributes(
    knode1_attributes: List[dict],
    knode2_attributes: List[dict],
    in_place: bool = False,
) -> List[dict]:

```

At least that's what I'm getting when i run it through `black --line-length=80`

Reply:
Is that compatible with this format (starts on line 219)?
```
        new_nodes[knode_key]["attributes"] = merge_attributes(
                                                new_nodes[knode_key]["attributes"],
                                                knode_value["attributes"],
                                                in_place=True
                                             )
```
or do I need to change something here as well

Reply:
Running that through `black` I get

```
  def merge_knodes(knodes1: dict, knodes2: dict) -> dict:
      """To merge knodes the keys must be the same. The knode values are merged
      by finding the union of their categories list and the union of their
      attributes list."""
      new_nodes = copy.deepcopy(knodes1)
      for knode_key, knode_value in knodes2.items():
          if knode_key not in new_nodes:
              new_nodes[knode_key] = copy.deepcopy(knode_value)
              continue
          new_nodes[knode_key]["attributes"] = merge_attributes(
              new_nodes[knode_key]["attributes"],
              knode_value["attributes"],
              in_place=True,
          )
          new_nodes[knode_key]["categories"] = merge_iterables(
              new_nodes[knode_key]["categories"], knode_value["categories"]
          )
          new_nodes[knode_key]["categories"].sort()
      return new_nodes
```

So essentially lining up function closing parens with the start of the function call + trying to fit as many args into 80 chars as possible. This is all exactly compliant with PEP8 but may uncompliant with some style guide that is specific to this project or to translator. 

Is there a style guide i should be referring to @patrickkwang ?

Reply:
No, we don't have a shared style guide. What black is doing looks good to me.

Reply:
Sounds good, I will go with that then!